src/Adapter/Security/CommandHandler/DeleteEmployeeSessionHandler.php

```
@@ -0,0 +1,65 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Security\CommandHandler;
+
+use EmployeeSession;
+use PrestaShop\PrestaShop\Core\Domain\Security\Command\DeleteEmployeeSessionCommand;
+use PrestaShop\PrestaShop\Core\Domain\Security\CommandHandler\DeleteEmployeeSessionHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\FailedToDeleteSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShopException;
+
+/**
+ * Class DeleteEmployeeSessionHandler
+ *
+ * @internal
+ */
+final class DeleteEmployeeSessionHandler implements DeleteEmployeeSessionHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DeleteEmployeeSessionCommand $command)
+    {
+        $sessionId = $command->getEmployeeSessionId()->getValue();
+
+        try {
+            $entity = new EmployeeSession($sessionId);
+
+            if ($entity->id != $sessionId) {
+                throw new SessionNotFoundException(sprintf('Session with id %s cannot be found.', var_export($sessionId, true)));
+            }
+
+            if (false === $entity->delete()) {
+                throw new FailedToDeleteSessionException(sprintf('Failed to delete Session with id %s', var_export($sessionId, true)));
+            }
+        } catch (PrestaShopException $e) {
+            throw new SessionException(sprintf('Unexpected error occurred when deleting Session with id %s', var_export($sessionId, true)), 0, $e);
```

Another thing. 
We've agreed (with @jolelievre & I suppose others) that its logical to throw CoreException wherever PrestashopException is thrown (instead of some domain exception)

---

So, what should I do, remove the try catch and let the PrestaShopException be raised?

---

instead of SessionException, throw CoreException